cmd/registry/scoring/expression.go

```diff
@@ -0,0 +1,145 @@
+	if err != nil {
+		return nil, fmt.Errorf("program construction error, %q: %s", expression, err)
+	}
+
+	out, _, err := prg.Eval(structProto.AsMap())
+	if err != nil {
+		return nil, fmt.Errorf("error in evaluating expression %q: %s", expression, err)
+	}
+
+	switch out.Type().TypeName() {
+	case "int":
+		expectedType := nativeTypeMap["int"]
+		nativeVal, err := out.ConvertToNative(expectedType)
+		if err != nil {
+			return nil, fmt.Errorf("failed converting output from type %s to %s: %s", out.Type().TypeName(), expectedType, err)
+		}
+		return nativeVal, nil
+	case "double":
+		expectedType := nativeTypeMap["double"]
+		nativeVal, err := out.ConvertToNative(expectedType)
+		if err != nil {
+			return nil, fmt.Errorf("failed converting output from type %s to %s: %s", out.Type().TypeName(), expectedType, err)
+		}
+		return nativeVal, nil
+	case "bool":
+		expectedType := nativeTypeMap["bool"]
+		nativeVal, err := out.ConvertToNative(expectedType)
+		if err != nil {
+			return nil, fmt.Errorf("failed converting output from type %s to %s: %s", out.Type().TypeName(), expectedType, err)
+		}
+		return nativeVal, nil
+	default:
+		return nil, fmt.Errorf("evaluating expression %q generated an unexpected output type %s: should be one of [int, float, bool]", expression, out.Type().TypeName())
+
+	}
+}
+
+func getStructProto(contents []byte, mimeType string) (*structpb.Struct, error) {
+	messageType, err := core.MessageTypeForMimeType(mimeType)
+	if err != nil {
+		return nil, fmt.Errorf("failed extracting message type from %q", mimeType)
+	}
+
+	switch messageType {
+	case "gnostic.metrics.Complexity":
+		return unmarshalAndStruct(contents, &metrics.Complexity{})
+	case "gnostic.metrics.Vocabulary":
+		return unmarshalAndStruct(contents, &metrics.Vocabulary{})
+	case "google.cloud.apigeeregistry.applications.v1alpha1.ConformanceReport":
+		return unmarshalAndStruct(contents, &rpc.ConformanceReport{})
+	case "google.cloud.apigeeregistry.applications.v1alpha1.Index":
+		return unmarshalAndStruct(contents, &rpc.Index{})
+	case "google.cloud.apigeeregistry.applications.v1alpha1.Lint":
+		return unmarshalAndStruct(contents, &rpc.Lint{})
+	case "google.cloud.apigeeregistry.v1.apihub.ReferenceList":
+		return unmarshalAndStruct(contents, &rpc.ReferenceList{})
+	case "google.cloud.apigeeregistry.v1.controller.Receipt":
+		return unmarshalAndStruct(contents, &rpc.Receipt{})
+	case "google.cloud.apigeeregistry.applications.v1alpha1.References":
+		return unmarshalAndStruct(contents, &rpc.References{})
+	case "google.cloud.apigeeregistry.v1.scoring.Score":
+		return unmarshalAndStruct(contents, &rpc.Score{})
+	case "google.cloud.apigeeregistry.v1.scoring.ScoreCard":
+		return unmarshalAndStruct(contents, &rpc.ScoreCard{})
+	// TODO: Add support for JSON artifacts
+	default:
+		return nil, fmt.Errorf("unsupported artifact type: %s", messageType)
+	}
+}
+
+func unmarshalAndStruct(contents []byte, message proto.Message) (*structpb.Struct, error) {
```

Comment:
It doesn't look like we use the `Struct` type for anything except converting it into a map. Can we update this to return `map[string]interface{}`? 

We can replace the protojson unmarshal step with a regular JSON unmarshal to skip converting to/from the struct type if our end goal is the map. I think working with maps will make the tests a lot cleaner too.

Reply:
Thanks for this suggestion. Done!